### PR TITLE
Reorganize the file structure of the project

### PR DESCRIPTION
**Pull request by zprobst**
In order to accommodate some of the new proposed features, namely:

- #36 
- #39 
- #10 

It will be beneficial to reorganize the project a little better now that the future of the project is a little more clearly defined. 

Changes Include:

- moved `nodestream.value_providers` under `nodestream.pipeline`.
- moved `nodestream.argument_resolvers` unser `nodestream.pipeline`
- moved `nodestream.normalizers` unser `nodestream.pipeline`
- consolidated extractor code under `nodestream.pipeline.extractors`
- moved `nodestream.audits` under `nodestream.project`
- moved `IngestStrategy` from `nodestream.model` to `nodestream.databases`
- moved indexes and schema from `nodestream.model` to `nodestream.schema`
- moved `nodestream.cli.schema_printers` to `nodestream.schema.printers`
- moved `nodestream.utilities` members to places where they are used 
- moved `nodestream.exceptions` members to places where they are used
- renamed `InterpreterContext` to `ProviderContext` to reflect its more broader use.

### CODE REVIEW

File: nodestream/project/pipeline_scope.py [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+from ..schema.schema import (
     AggregatedIntrospectiveIngestionComponent,
     IntrospectiveIngestionComponent,
 )
 from .pipeline_definition import PipelineDefinition
 from .run_request import RunRequest
 
 
+class MissingExpectedPipelineError(ValueError):
+    pass
```
Comment (ccloes) [inline nodestream/project/pipeline_scope.py:13]: Is this intended to be a pass?

**Review by ccloes — APPROVED** [review]
Didn't see any errors in the move.  A couple comments just to make sure those were intended changes as well.

File: nodestream/cli/operations/generate_pipeline_scaffold.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
 from pathlib import Path
-from typing import Iterable
+from typing import Any, Iterable
```
[the remark this answers is not in the document]
Comment (zprobst) [reply nodestream/cli/operations/generate_pipeline_scaffold.py:2]: This was needed for moving `pretty_print_yaml_to_file` out of `utilities.py`

File: nodestream/pipeline/extractors/extractor.py [hunk 3]
```diff
@@ -19,18 +19,3 @@ def handle_async_record_stream(
     @abstractmethod
     async def extract_records(self) -> AsyncGenerator[Any, Any]:
         raise NotImplementedError
-
-
-class IterableExtractor(Extractor):
```
[the remark this answers is not in the document]
Comment (zprobst) [reply nodestream/pipeline/extractors/extractor.py:24]: Yes this found a new hope in `nodestream.pipeline.extractors.iterable`

File: nodestream/pipeline/extractors/extractor.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from abc import abstractmethod
-from typing import Any, AsyncGenerator, Iterable
+from typing import Any, AsyncGenerator
```
[the remark this answers is not in the document]
Comment (zprobst) [reply nodestream/pipeline/extractors/extractor.py:2]: See below for clarification

File: nodestream/pipeline/extractors/iterable.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+from typing import Any, AsyncGenerator, Iterable
```
[the remark this answers is not in the document]
Comment (zprobst) [reply nodestream/pipeline/extractors/iterable.py:1]: ha

File: nodestream/project/pipeline_scope.py (hunk 1 above)
Comment (zprobst) [reply nodestream/project/pipeline_scope.py:13]: Yeah just providing a specific type for the error .

**Pull request merged**